src/libraries/System.Data.Common/ref/System.Data.Common.cs

```diff
@@ -1943,17 +1961,18 @@ public partial class DbConnectionStringBuilder : System.Collections.ICollection,
         [System.ComponentModel.BrowsableAttribute(false)]
         public bool IsReadOnly { get { throw null; } }
         [System.ComponentModel.BrowsableAttribute(false)]
+        [System.Diagnostics.CodeAnalysis.AllowNull]
         public virtual object this[string keyword] { get { throw null; } set { } }
```

Comment:
Oracle's MySQL Connector/NET has several keys that do return `null`, e.g., from `new MySqlConnectionStringBuilder()["SslKey"]`. (I'm fixing this for MySqlConnector: https://github.com/mysql-net/MySqlConnector/issues/837.)

This is also inconsistent with `object? System.Collections.IDictionary.this[object keyword]` (below), but I'm assuming that annotation is required because of the `IDictionary` interface?

I think it probably is better overall to coerce connection string properties to non-null (when unset) but it's not consistently done in existing drivers.

Reply:
> This is also inconsistent with object? System.Collections.IDictionary.this[object keyword] (below), but I'm assuming that annotation is required because of the IDictionary interface?

Yes, this is only because of the IDictionary interface. So if you're using the explicit interface methods, they're nullable (because they must be), but if you're using the regular methods they're not.

The general logic of DbConnectionStringBuilder is that setting to null removes the item, so yeah, I agree that making properties return non-nullable values makes sense. Thanks for your input on this.